site/utils/apiUtils/index.ts

```diff
@@ -107,3 +115,205 @@ export const getReturnPage = (req: NextApiRequest) => {
         ? queryParam
         : null;
 };
+
+export const delay = (ms: number) => new Promise((res) => setTimeout(res, ms));
+
+export const publishToHootsuite = async (socialMediaPosts: SocialMediaPost[], orgId: string) => {
+    await Promise.all(
+        socialMediaPosts
+            .filter((s) => s.status === SocialMediaPostStatus.pending)
+            .map(async (socialMediaPost) => {
+                const refreshTokens = await getParametersByPath(`/social/${orgId}/hootsuite`);
+
+                if (!refreshTokens || (refreshTokens && refreshTokens.Parameters?.length === 0)) {
+                    await upsertSocialMediaPost(
+                        {
+                            ...socialMediaPost,
+                            status: SocialMediaPostStatus.rejected,
+                        },
+                        orgId,
+                    );
```

Comment:
This can be refactored to another method which can be re-used

Reply:
it would still be the same length though no need to make another method